brave/browser-laptop
# Replace Payment History CSV export with Contribution Statement PDF

This is a Work-In-Progress PR, implementing #4769, so feel free to ignore, critique, fix, or replace with your own solution as you please.
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist. (issue #4769 )
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message. (This is WIP, will be rebased)
- [ ] Ran `git rebase -i` to squash commits (This is WIP, will be rebased)

Test Plan:
1. Open Brave
2. Open Preferences -> Payments, and ensure it is enabled
3. If you don't have a Payment History, you can generate a fake one at terminal with `npm run add-simulated-payment-history`
4. Click "Show Payment History"
5. Observe "Receipt Link" column now has links ending in ".pdf"
6. Click a receipt link, and confirm that a Save dialog opens and suggested filename is same as the one displayed in Receipt Link column
7. Click "Save", and observe that the PDF is downloaded and appears in the Downloads bar at bottom of screen
8. Double click the downloaded file
9. Check PDF against mockup
